File: backend/src/main/java/com/pokecollect/events/CardRemovedFromBinder.java
package com.pokecollect.events;

import java.time.Instant;
import java.util.UUID;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

/** Emitted when a binder pocket is cleared. */
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public record CardRemovedFromBinder(
    String userId,
    int pageNumber,
    int slotIndex,
    String eventType,
    String eventId,
    String timestamp
) {
    public static CardRemovedFromBinder of(String userId, int pageNumber, int slotIndex) {
        return new CardRemovedFromBinder(
            userId, pageNumber, slotIndex,
            "card_removed_from_binder", UUID.randomUUID().toString(), Instant.now().toString());
    }
}

File: backend/src/main/kotlin/com/tcgtracker/events/CardRemovedFromBinder.kt
package com.tcgtracker.events

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import java.time.Instant
import java.util.UUID

/** Emitted when a binder pocket is cleared. Snake_case JSON (see [CardAddedToCollection]). */
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class CardRemovedFromBinder(
    val userId: String,
    val pageNumber: Int,
    val slotIndex: Int,
    val eventType: String,
    val eventId: String,
    val timestamp: String,
) {
    companion object {
        @JvmStatic
        fun of(userId: String, pageNumber: Int, slotIndex: Int): CardRemovedFromBinder =
            CardRemovedFromBinder(
                userId, pageNumber, slotIndex,
                "card_removed_from_binder", UUID.randomUUID().toString(), Instant.now().toString(),
            )
    }
}
